#!/usr/bin/env python

PACKAGE = "bixi_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_area", 				int_t,    0, "An Integer parameter", 300,	0,10000)
#blue
gen.add("blue_H_low", 			int_t,    0, "An Integer parameter", 100,	0,179)
gen.add("blue_H_high", 			int_t,    0, "An Integer parameter", 170,	0,179)
gen.add("blue_S_low",  			int_t,    0, "An Integer parameter", 75,	0,255)
gen.add("blue_S_high", 			int_t,    0, "An Integer parameter", 180,	0,255)
gen.add("blue_V_low",  			int_t,    0, "An Integer parameter", 50,	0,255)
gen.add("blue_V_high", 			int_t,    0, "An Integer parameter", 160,	0,255)
# black
gen.add("black_H_low", 			int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("black_H_high", 		int_t,    0, "An Integer parameter", 255,	0,179)
gen.add("black_S_low",  		int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("black_S_high", 		int_t,    0, "An Integer parameter", 100,	0,255)
gen.add("black_V_low", 			int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("black_V_high",			int_t,    0, "An Integer parameter", 90,	0,255)
#canny/Hough operations
gen.add("cannyThreshold",		int_t,		0, "An Integer parameter", 20,	0,200)
gen.add("accumulatorThreshold",int_t,	0, "An Integer parameter", 70,  0,300)

exit(gen.generate(PACKAGE, "bixi_vision", "building_blocks"))